client/public/js/components/EventDetail.js

```diff
@@ -10,64 +10,70 @@ var EventDetail = React.createClass({
     return EventStore.getCurrentEvent();
   },
 
+  // Adds event listeners when events are created or edited
   componentDidMount: function() {
-    //adds an event listener for when events are created
     EventDetailStore.addEventListener('create', this._onCreate);
-    //adds an event listener for when current event is edited
     EventStore.addEventListener('edit', this._onEdit);
   },
 
+  // Removes event listener when events are deleted or edited
   componentWillUnmount: function() {
-    //removes an event listener for when events are deleted
     EventDetailStore.removeEventListener('create', this._onCreate);
-    //removes an event listener for when events are edited
     EventStore.removeEventListener('edit', this._onEdit);
   },
 
   render: function() {
     return (
       <div>
-
-      <p>Event: {this.state.name}
-      {this.state.mode === 'shepherd' ? <button onClick={this._editEvent}>Edit Event</button> : null}
-      </p>
-
-      <p>Location: {this.state.location}
-      {this.state.mode === 'shepherd' ? <button onClick={this._editLocation}>Edit Location</button> : null}
-      </p>
-
-      {this.state.mode === 'sheep' ? <ObservationCreate eventId={this.state.id}/> : null}
-
-      <ObservationList eventId={this.state.id}/>
-
+        <p>Event: {this.state.name}
+          {this.state.mode === 'shepherd' ? <button onClick={this._editEvent}>Edit Event</button> : null}
+        </p>
+        <p>Location: {this.state.location}
+          { this.state.mode === 'shepherd' ? <button onClick={this._editLocation}>Edit Location</button> : null}
+        </p>
+        { /* TODO: Set up API end point to track whether a user is in an event */ }
+        { /* Display Join or Leave event based on whether the sheep is currently in the event */ }
+        { (this.state.mode === 'sheep' && true) ? <button onClick={this._toggleJoin}>Leave Event</button> : null }
+        { (this.state.mode === 'sheep' && false) ? <button onClick={this._toggleJoin}>Join Event</button> : null }

```

Comment:
We can refactor this later to be something like the following:

`<button onClick={this._toggleJoin}>{(this.state.mode === 'sheep' && true) ? 'Leave' : 'Join'} Event</button>`